code/modules/codex/codex_cataloguer.dm

```diff
@@ -0,0 +1,294 @@
+	var/tmp/partial_scan_time = 0
+	/// Weakref of the thing that was last scanned if inturrupted. Used to allow for partial scans to be resumed.
+	var/tmp/weakref/partial_scanned
+	/// Set to a user mob when scanning, to stop multiple scans.
+	var/weakref/scanning
+	/// Dummy object for holding the scanning effect.
+	var/obj/screen/scan_radius/scan_radius_overlay
+	/// Holder for the currently scanning beam effect.
+	var/datum/beam/scan_beam
+	/// Currently loaded survey disk for storing Good Boy Points.
+	var/obj/item/disk/survey/loaded_disk
+
+/obj/item/cataloguer/Initialize(ml, material_key)
+	. = ..()
+	update_icon()
+	set_scan_radius(scan_range, TRUE)
+
+/obj/item/cataloguer/Destroy()
+	QDEL_NULL(loaded_disk)
+	QDEL_NULL(scan_beam)
+	stop_scan()
+	scanning = null
+	QDEL_NULL(scan_radius_overlay)
+	return ..()
+
+/obj/item/cataloguer/physically_destroyed(skip_qdel)
+	if(loaded_disk)
+		loaded_disk.dropInto(loc)
+		loaded_disk = null
+	return ..()
+
+/obj/item/cataloguer/mapped/Initialize(ml, material_key)
+	. = ..()
+	loaded_disk = new(src)
+
+/obj/item/cataloguer/proc/set_scan_radius(var/new_range, var/forced)
+	if(!forced && new_range == scan_range)
+		return FALSE
+
+	scan_range = new_range
+	QDEL_NULL(scan_radius_overlay)
+	scan_radius_overlay = new
+	scan_radius_overlay.color = glow_colour
+	scan_radius_overlay.set_radius(new_range)
+	return TRUE
+
+/obj/item/cataloguer/on_update_icon()
+	icon_state = get_world_inventory_state()
+	cut_overlays()
+	if(scanning && check_state_in_icon("[icon_state]_glow", icon))
+		var/image/I
+		if(plane == HUD_PLANE)
+			I = image(icon, "[icon_state]_glow")
+		else
+			I = emissive_overlay(icon, "[icon_state]_glow")
+		I.appearance_flags |= RESET_COLOR
+		I.color = glow_colour
+		add_overlay(I)
+
+/obj/item/cataloguer/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(scanning)
+		to_chat(user, SPAN_WARNING("\The [src] is already scanning something."))
+		return
+	start_scan(target, user)
+
+/obj/item/cataloguer/attack_self(mob/user)
+	if(loaded_disk)
+		to_chat(user, SPAN_NOTICE("You pop \the [loaded_disk] out of \the [src]."))
+		loaded_disk.dropInto(loc)
+		user.put_in_hands(loaded_disk)
+		loaded_disk = null
+		playsound(user.loc, 'sound/weapons/flipblade.ogg', 50, 1)
+		return TRUE
+	return ..()
+	
+/obj/item/cataloguer/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/disk/survey))
+		if(loaded_disk)
+			to_chat(user, SPAN_WARNING("\The [src] already has a disk loaded."))
+		else if(user.unEquip(W, src))
+			loaded_disk = W
+			playsound(user.loc, 'sound/weapons/flipblade.ogg', 50, 1)
+			to_chat(user, SPAN_NOTICE("You slot \the [W] into \the [src]."))
+		return TRUE
+	return ..()
+
+/obj/item/cataloguer/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	if(loaded_disk)
+		to_chat(user, "It has \a [loaded_disk] slotted into the storage port. The display indicates it currently holds [loaded_disk.data] Good Explorer Point\s.")
```

Comment:
Willing to change this away from a GBP reference, just using the same language as the pre-existing explo point scheme.